sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java

```diff
@@ -123,15 +140,29 @@ public ProviderResponse changePassword(
         ProviderResponse providerResponse =
             storeProvider
                 .getWriterStore(realm, userStorage)
-                .changePassword(userId, oldPassword, newPassword, providerRequest);
+                .changePassword(
+                    userId,
+                    oldPassword,
+                    newPassword,
+                    webserviceTag,
+                    templateProperties,
+                    providerRequest);
         sugoiEventPublisher.publishCustomEvent(
             realm,
             userStorage,
             SugoiEventTypeEnum.CHANGE_PASSWORD,
             Map.ofEntries(
                 Map.entry(EventKeysConfig.NEW_PASSWORD, newPassword),
                 Map.entry(EventKeysConfig.OLD_PASSWORD, newPassword),
-                Map.entry(EventKeysConfig.USER, userId)));
+                Map.entry(EventKeysConfig.USER_ID, userId),
+                Map.entry(EventKeysConfig.PROPERTIES, templateProperties),
+                Map.entry(
+                    EventKeysConfig.MAIL,
+                    templateProperties.get("mail") != null
+                        ? templateProperties.get("mail")
+                        : (user.getMail() != null ? user.getMail() : "")),
+                Map.entry(
+                    EventKeysConfig.WEBSERVICE_TAG, webserviceTag != null ? webserviceTag : "")));
```

Comment:
```
               Map.entry(EventKeysConfig.WEBSERVICE_TAG, webhookTag != null ? webhookTag : "MAIL"),
```
No ?

Reply:
I thought that MAIL should better be added later, if you're okay with this I'll better remove "MAIL" at the other location in this file

Reply:
Indeed we can do like this !